.github/CODEOWNERS

```diff
@@ -1,12 +1,67 @@
 # Codeowners for IREE Github Repository.
 # Refer to https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners
-# for syntax of this file.
+# for syntax of this file (tl;dr: syntax is like .gitignore. Last matching rule
+# takes precedence).
+# Because of the precedence, rules for directories are listed topologically.
 
-# Global code owners
+# No global owners because we don't really want e.g. changing the root
+# CMakeLists.txt file to always ping a bunch of people.
 
 # Code owners for individual components/directories
+
+# Third-Party Code
+/.gitmodules @GMNGeoffrey @ScottTodd @stellaraccident
+/third_party/ @GMNGeoffrey @ScottTodd @stellaraccident
+
+# Bindings
+/bindings/python/ @stellaraccident
+/bindings/tflite/ @benvanik
+
+# Integrations
+/integrations/ @benvanik @stellaraccident
+/integrations/tensorflow/ @stellaraccident
+/integrations/tensorflow/build_tools/ @GMNGeoffrey @stellaraccident
+/integrations/tensorflow/test/**/iree_tfl_tests/ @rsuderman
+
+# Experimental
+# It's experimental, but we still don't want any old directory added here.
+/experimental/ @benvanik @GMNGeoffrey @stellaraccident
+/experimental/remoting/ @stellaraccident
+/experimental/rocm/ @ThomasRaoux
+/experimental/web/ @ScottTodd
+
+# Infra Top-Level Directories
+/benchmarks/ @GMNGeoffrey @antiagainst
+/build_tools/ @GMNGeoffrey @ScottTodd
+/build_tools/benchmarks/ @GMNGeoffrey @antiagainst
+/build_tools/python_deploy/ @stellaraccident
+/build_tools/third_party/ @GMNGeoffrey @ScottTodd @stellaraccident
+/.github/ @GMNGeoffrey @ScottTodd
+/scripts/ @GMNGeoffrey @ScottTodd
+
+# Other Top-Level Directories
+/colab/ @ScottTodd
+/docs/ @ScottTodd
+/llvm-external-projects/ @stellaraccident
+
+# Main IREE directories
+/iree/ @benvanik
+
+# Compiler
 /iree/compiler/Codegen/ @MaheshRavishankar
 /iree/compiler/Codegen/LLVMCPU/ @MaheshRavishankar @hanhanW
 /iree/compiler/Codegen/LLVMGPU/ @MaheshRavishankar @ThomasRaoux
 /iree/compiler/Codegen/Sandbox/ @MaheshRavishankar @hanhanW
 /iree/compiler/Codegen/SPIRV/ @MaheshRavishankar @antiagainst
+/iree/compiler/ConstEval/ @stellaraccident
+/iree/compiler/InputConversion/ @stellaraccident
+
```

Comment:
Can add me to https://github.com/google/iree/tree/main/iree/compiler/Dialect/Vulkan too.